craft_parts/sources/git_source.py

```diff
@@ -174,11 +174,25 @@ def __init__(  # pylint: disable=too-many-arguments
     ) -> None:
         super().__init__(source, part_src_dir, **kwargs)
 
+    def _expand_commit(self, commit: str) -> str:
+        """Expand a commit hash to full length."""
+        # 'git rev-parse' expands non-ambiguous commits to full length (40 characters).
+        # Full length commits are unchanged.
+        return self._run_output(
+            [
+                get_git_command(),
+                "-C",
+                str(self.part_src_dir),
+                "rev-parse",
```

Comment:
A consideration, possibly for near-future work; if `source-depth` is non-zero, this call will probably fail:

```
❯ git rev-parse cd483f3
cd483f3
fatal: ambiguous argument 'cd483f3': unknown revision or path not in the working tree.
```

Should we handle this in any particular way?

Reply:
In particular, the case that will probably be common is trying to shallow-clone to a short-hash

Reply:
I like this idea, it also addresses the case where there's a typo on the short hash
but I'm also ok with doing this a bit later if you prefer

Reply:
Added :)